devito/finite_differences/finite_difference.py

```diff
@@ -1,15 +1,17 @@
 from __future__ import absolute_import
 
-from functools import reduce
+from functools import reduce, partial
 from operator import mul
 
-from sympy import finite_diff_weights, cos, sin
+from sympy import finite_diff_weights
 
-from devito.logger import error
+from devito.logger import error, debug
+from devito.finite_differences.operations import Add
 
 __all__ = ['first_derivative', 'second_derivative', 'cross_derivative',
            'generic_derivative', 'second_cross_derivative',
-           'left', 'right', 'centered', 'staggered_diff']
+           'left', 'right', 'centered', 'staggered_diff',
```

Comment:
do we actually need ``left/right/center`` outside of this module?

Reply:
`first_derivative` is exported as a general purpose FD function and needs it so I'd say yes